File: 01-Beginner-Client-libraries/08-Implementing-custom-interfaces/ros2_ws/src/more_interfaces/scripts/publish_contact.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from more_interfaces.msg import Contact

class ContactPublisher(Node):

    def __init__(self):
        super().__init__('address_book_publisher_py')
        self.publisher_ = self.create_publisher(Contact, 'address_book', 10)
        self.timer_     = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        msg              = Contact()
        msg.first_name   = 'John'
        msg.last_name    = 'Doe'
        msg.phone_number = '1234567890'
        msg.phone_type   = Contact.PHONE_TYPE_MOBILE
        self.get_logger().info(f'Publishing: {msg.first_name} {msg.last_name}')
        self.publisher_.publish(msg)


def main():
    rclpy.init()
    node = ContactPublisher()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: 01-Beginner-Client-libraries/08-Implementing-custom-interfaces/ros2_ws/src/more_interfaces/src/publish_contact.cpp
#include <chrono>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "more_interfaces/msg/contact.hpp"

using namespace std::chrono_literals;


class ContactPublisher : public rclcpp::Node {
    private:
        rclcpp::Publisher<more_interfaces::msg::Contact>::SharedPtr contact_publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
    public:
        ContactPublisher() : Node("contact_publisher") {
            contact_publisher_ = this->create_publisher<more_interfaces::msg::Contact>("contact", 10);
            auto publish_msg = [this] () -> void {
                auto msg = more_interfaces::msg::Contact();
                msg.first_name   = "AX";
                msg.last_name    = "BY";
                msg.phone_number = "1234567890";
                msg.phone_type   = msg.PHONE_TYPE_MOBILE;
                RCLCPP_INFO(this->get_logger(), "Publishing: %s %s", msg.first_name.c_str(), msg.last_name.c_str());
                this->contact_publisher_->publish(msg);
            };
            timer_ = this->create_wall_timer(1s, publish_msg);
        }
};


int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<ContactPublisher>());
    rclcpp::shutdown();
    return 0;
}